runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -704,6 +709,28 @@ TR_VectorAPIExpansion::expandVectorAPI()
       TR::Node *parent = NULL;
       TR::MethodSymbol *methodSymbol = NULL;
 
+
+      // remove unnecessary checkcast and NULLCHK as soon as possible
+      if (node->getOpCodeValue() == TR::checkcast ||
+         node->getOpCodeValue() == TR::NULLCHK)
+         {
+         TR::Node *firstChild = node->getFirstChild();
+         if (firstChild->getOpCodeValue() == TR::PassThrough)
+            firstChild = firstChild->getFirstChild();
+
+         if (firstChild->getOpCode().hasSymbolReference())
+            {
+            int32_t childClassId = _aliasTable[firstChild->getSymbolReference()->getReferenceNumber()]._classId;
+            if (childClassId > 0)
+               {
+               // can be removed since the child will be scalarized or vectorized
+               TR::Node::recreate(node, TR::treetop);
```

Comment:
If `node` is a `checkcast`, this will result in a `treetop` node with two children

Reply:
Good point! I'll just set number of children to 1.

Reply:
Would you mind anchoring the second child? I believe that currently it has to be a `loadaddr`, but we might start allowing non-constant cast types like we do for `instanceof`